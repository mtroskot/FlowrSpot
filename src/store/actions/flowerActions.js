import { flowerActionTypes } from 'src/constants/actionTypes';

export const getFavoriteFlowers = () => ({
  type: flowerActionTypes.GET_FAVORITE_FLOWERS,
  payload: {}
});

export const getFavoriteFlowersSuccess = favoriteFlowers => ({
  type: flowerActionTypes.GET_FAVORITE_FLOWERS_SUCCESS,
  payload: { favoriteFlowers }
});

export const favoriteFlower = flowerId => ({
  type: flowerActionTypes.FAVORITE_FLOWER,
  payload: { flowerId }
});

export const favoriteFlowerSuccess = (flowerId, favoriteId) => ({
  type: flowerActionTypes.FAVORITE_FLOWER_SUCCESS,
  payload: { flowerId, favoriteId }
});

export const unfavoriteFlower = (flowerId, favoriteId) => ({
  type: flowerActionTypes.UNFAVORITE_FLOWER,
  payload: { flowerId, favoriteId }
});

export const unfavoriteFlowerSuccess = flowerId => ({
  type: flowerActionTypes.UNFAVORITE_FLOWER_SUCCESS,
  payload: { flowerId }
});
