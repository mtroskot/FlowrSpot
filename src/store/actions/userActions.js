import { userActionTypes } from 'src/constants/actionTypes';

export const registerUser = registerData => ({
  type: userActionTypes.REGISTER_USER,
  payload: { registerData }
});

export const loginUser = (email, password) => ({
  type: userActionTypes.LOGIN_USER,
  payload: { email, password }
});

export const loginUserSuccess = (id, email, authToken, firstName, lastName) => ({
  type: userActionTypes.LOGIN_USER_SUCCESS,
  payload: { id, email, authToken, firstName, lastName }
});

export const logoutUser = () => ({
  type: userActionTypes.LOGOUT_USER,
  payload: {}
});
