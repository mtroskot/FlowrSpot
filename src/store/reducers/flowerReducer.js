import { flowerActionTypes } from 'src/constants/actionTypes';

const initialState = {
  favorites: []
};

const flowerReducer = (state = initialState, { type, payload }) => {
  switch (type) {
    case flowerActionTypes.GET_FAVORITE_FLOWERS_SUCCESS:
      return {
        ...state,
        favorites: [...state.favorites, ...payload.favoriteFlowers]
      };
    case flowerActionTypes.FAVORITE_FLOWER_SUCCESS:
      return {
        ...state,
        favorites: [...state.favorites, { flowerId: payload.flowerId, favoriteId: payload.favoriteId }]
      };
    case flowerActionTypes.UNFAVORITE_FLOWER_SUCCESS:
      return {
        ...state,
        favorites: state.favorites.filter(favorite => favorite.flowerId !== payload.flowerId)
      };
    default:
      return state;
  }
};

export default flowerReducer;
