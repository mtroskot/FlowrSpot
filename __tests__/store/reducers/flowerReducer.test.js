import flowerReducer from 'src/store/reducers/flowerReducer';
import * as flowerActions from 'src/store/actions/flowerActions';

const initialState = {
  favorites: []
};

describe('INITIAL_STATE', () => {
  const beforeActionState = initialState;

  it('1returns initial state', () => {
    const action = { type: 'dummy_action' };
    const expectedState = beforeActionState;
    expect(flowerReducer(beforeActionState, action)).toEqual(expectedState);
    expect(flowerReducer(beforeActionState, action)).toMatchSnapshot();
  });

  it('2returns getFavoriteFlowersSuccess state', () => {
    const favoriteFlowers = [
      { flowerId: 1, favoriteId: 1 },
      { flowerId: 2, favoriteId: 2 },
      { flowerId: 3, favoriteId: 3 }
    ];
    const getFavoriteFlowerAction = flowerActions.getFavoriteFlowersSuccess(favoriteFlowers);
    const expectedState = { ...beforeActionState, favorites: [...beforeActionState.favorites, ...favoriteFlowers] };
    expect(flowerReducer(beforeActionState, getFavoriteFlowerAction)).toEqual(expectedState);
    expect(flowerReducer(beforeActionState, getFavoriteFlowerAction)).toMatchSnapshot();
  });

  it('3returns favoriteFlowerSuccess state', () => {
    const flowerId = 1;
    const favoriteId = 1;
    const favoriteFlowerAction = flowerActions.favoriteFlowerSuccess(flowerId, favoriteId);
    const expectedState = {
      ...beforeActionState,
      favorites: [...beforeActionState.favorites, { flowerId, favoriteId }]
    };
    expect(flowerReducer(beforeActionState, favoriteFlowerAction)).toEqual(expectedState);
    expect(flowerReducer(beforeActionState, favoriteFlowerAction)).toMatchSnapshot();
  });

  it('4returns unfavoriteFlowerSuccess state', () => {
    const flowerId = 1;
    const getFavoriteFlowerAction = flowerActions.unfavoriteFlowerSuccess(flowerId);
    const expectedState = {
      ...beforeActionState,
      favorites: beforeActionState.favorites.filter(favorite => favorite.flowerId !== flowerId)
    };
    expect(flowerReducer(beforeActionState, getFavoriteFlowerAction)).toEqual(expectedState);
    expect(flowerReducer(beforeActionState, getFavoriteFlowerAction)).toMatchSnapshot();
  });
});

describe('GET_FAVORITE_FLOWERS_SUCCESS STATE', () => {
  const beforeActionState = {
    ...initialState,
    favorites: [{ flowerId: 1, favoriteId: 1 }, { flowerId: 2, favoriteId: 2 }, { flowerId: 3, favoriteId: 3 }]
  };

  it('1returns initial state', () => {
    const action = { type: 'dummy_action' };
    const expectedState = beforeActionState;
    expect(flowerReducer(beforeActionState, action)).toEqual(expectedState);
    expect(flowerReducer(beforeActionState, action)).toMatchSnapshot();
  });

  it('2returns getFavoriteFlowersSuccess state', () => {
    const favoriteFlowers = [
      { flowerId: 4, favoriteId: 4 },
      { flowerId: 5, favoriteId: 5 },
      { flowerId: 6, favoriteId: 6 }
    ];
    const getFavoriteFlowerAction = flowerActions.getFavoriteFlowersSuccess(favoriteFlowers);
    const expectedState = { ...beforeActionState, favorites: [...beforeActionState.favorites, ...favoriteFlowers] };
    expect(flowerReducer(beforeActionState, getFavoriteFlowerAction)).toEqual(expectedState);
    expect(flowerReducer(beforeActionState, getFavoriteFlowerAction)).toMatchSnapshot();
  });

  it('3returns favoriteFlowerSuccess state', () => {
    const flowerId = 7;
    const favoriteId = 7;
    const favoriteFlowerAction = flowerActions.favoriteFlowerSuccess(flowerId, favoriteId);
    const expectedState = {
      ...beforeActionState,
      favorites: [...beforeActionState.favorites, { flowerId, favoriteId }]
    };
    expect(flowerReducer(beforeActionState, favoriteFlowerAction)).toEqual(expectedState);
    expect(flowerReducer(beforeActionState, favoriteFlowerAction)).toMatchSnapshot();
  });

  it('4returns unfavoriteFlowerSuccess state', () => {
    const flowerId = 1;
    const getFavoriteFlowerAction = flowerActions.unfavoriteFlowerSuccess(flowerId);
    const expectedState = {
      ...beforeActionState,
      favorites: beforeActionState.favorites.filter(favorite => favorite.flowerId !== flowerId)
    };
    expect(flowerReducer(beforeActionState, getFavoriteFlowerAction)).toEqual(expectedState);
    expect(flowerReducer(beforeActionState, getFavoriteFlowerAction)).toMatchSnapshot();
  });
});

describe('FAVORITE_FLOWER_SUCCESS STATE', () => {
  const beforeActionState = {
    ...initialState,
    favorites: [{ flowerId: 1, favoriteId: 1 }]
  };

  it('1returns initial state', () => {
    const action = { type: 'dummy_action' };
    const expectedState = beforeActionState;
    expect(flowerReducer(beforeActionState, action)).toEqual(expectedState);
    expect(flowerReducer(beforeActionState, action)).toMatchSnapshot();
  });

  it('2returns getFavoriteFlowersSuccess state', () => {
    const favoriteFlowers = [
      { flowerId: 4, favoriteId: 4 },
      { flowerId: 5, favoriteId: 5 },
      { flowerId: 6, favoriteId: 6 }
    ];
    const getFavoriteFlowerAction = flowerActions.getFavoriteFlowersSuccess(favoriteFlowers);
    const expectedState = { ...beforeActionState, favorites: [...beforeActionState.favorites, ...favoriteFlowers] };
    expect(flowerReducer(beforeActionState, getFavoriteFlowerAction)).toEqual(expectedState);
    expect(flowerReducer(beforeActionState, getFavoriteFlowerAction)).toMatchSnapshot();
  });

  it('3returns favoriteFlowerSuccess state', () => {
    const flowerId = 7;
    const favoriteId = 7;
    const favoriteFlowerAction = flowerActions.favoriteFlowerSuccess(flowerId, favoriteId);
    const expectedState = {
      ...beforeActionState,
      favorites: [...beforeActionState.favorites, { flowerId, favoriteId }]
    };
    expect(flowerReducer(beforeActionState, favoriteFlowerAction)).toEqual(expectedState);
    expect(flowerReducer(beforeActionState, favoriteFlowerAction)).toMatchSnapshot();
  });

  it('4returns unfavoriteFlowerSuccess state', () => {
    const flowerId = 1;
    const getFavoriteFlowerAction = flowerActions.unfavoriteFlowerSuccess(flowerId);
    const expectedState = {
      ...beforeActionState,
      favorites: beforeActionState.favorites.filter(favorite => favorite.flowerId !== flowerId)
    };
    expect(flowerReducer(beforeActionState, getFavoriteFlowerAction)).toEqual(expectedState);
    expect(flowerReducer(beforeActionState, getFavoriteFlowerAction)).toMatchSnapshot();
  });
});
