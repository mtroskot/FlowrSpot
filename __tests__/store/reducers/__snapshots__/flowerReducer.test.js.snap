// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FAVORITE_FLOWER_SUCCESS STATE 1returns initial state 1`] = `
Object {
  "favorites": Array [
    Object {
      "favoriteId": 1,
      "flowerId": 1,
    },
  ],
}
`;

exports[`FAVORITE_FLOWER_SUCCESS STATE 2returns getFavoriteFlowersSuccess state 1`] = `
Object {
  "favorites": Array [
    Object {
      "favoriteId": 1,
      "flowerId": 1,
    },
    Object {
      "favoriteId": 4,
      "flowerId": 4,
    },
    Object {
      "favoriteId": 5,
      "flowerId": 5,
    },
    Object {
      "favoriteId": 6,
      "flowerId": 6,
    },
  ],
}
`;

exports[`FAVORITE_FLOWER_SUCCESS STATE 3returns favoriteFlowerSuccess state 1`] = `
Object {
  "favorites": Array [
    Object {
      "favoriteId": 1,
      "flowerId": 1,
    },
    Object {
      "favoriteId": 7,
      "flowerId": 7,
    },
  ],
}
`;

exports[`FAVORITE_FLOWER_SUCCESS STATE 4returns unfavoriteFlowerSuccess state 1`] = `
Object {
  "favorites": Array [],
}
`;

exports[`GET_FAVORITE_FLOWERS_SUCCESS STATE 1returns initial state 1`] = `
Object {
  "favorites": Array [
    Object {
      "favoriteId": 1,
      "flowerId": 1,
    },
    Object {
      "favoriteId": 2,
      "flowerId": 2,
    },
    Object {
      "favoriteId": 3,
      "flowerId": 3,
    },
  ],
}
`;

exports[`GET_FAVORITE_FLOWERS_SUCCESS STATE 2returns getFavoriteFlowersSuccess state 1`] = `
Object {
  "favorites": Array [
    Object {
      "favoriteId": 1,
      "flowerId": 1,
    },
    Object {
      "favoriteId": 2,
      "flowerId": 2,
    },
    Object {
      "favoriteId": 3,
      "flowerId": 3,
    },
    Object {
      "favoriteId": 4,
      "flowerId": 4,
    },
    Object {
      "favoriteId": 5,
      "flowerId": 5,
    },
    Object {
      "favoriteId": 6,
      "flowerId": 6,
    },
  ],
}
`;

exports[`GET_FAVORITE_FLOWERS_SUCCESS STATE 3returns favoriteFlowerSuccess state 1`] = `
Object {
  "favorites": Array [
    Object {
      "favoriteId": 1,
      "flowerId": 1,
    },
    Object {
      "favoriteId": 2,
      "flowerId": 2,
    },
    Object {
      "favoriteId": 3,
      "flowerId": 3,
    },
    Object {
      "favoriteId": 7,
      "flowerId": 7,
    },
  ],
}
`;

exports[`GET_FAVORITE_FLOWERS_SUCCESS STATE 4returns unfavoriteFlowerSuccess state 1`] = `
Object {
  "favorites": Array [
    Object {
      "favoriteId": 2,
      "flowerId": 2,
    },
    Object {
      "favoriteId": 3,
      "flowerId": 3,
    },
  ],
}
`;

exports[`INITIAL_STATE 1returns initial state 1`] = `
Object {
  "favorites": Array [],
}
`;

exports[`INITIAL_STATE 2returns getFavoriteFlowersSuccess state 1`] = `
Object {
  "favorites": Array [
    Object {
      "favoriteId": 1,
      "flowerId": 1,
    },
    Object {
      "favoriteId": 2,
      "flowerId": 2,
    },
    Object {
      "favoriteId": 3,
      "flowerId": 3,
    },
  ],
}
`;

exports[`INITIAL_STATE 3returns favoriteFlowerSuccess state 1`] = `
Object {
  "favorites": Array [
    Object {
      "favoriteId": 1,
      "flowerId": 1,
    },
  ],
}
`;

exports[`INITIAL_STATE 4returns unfavoriteFlowerSuccess state 1`] = `
Object {
  "favorites": Array [],
}
`;
